corehq/ex-submodules/pillowtop/processors/elastic.py

```diff
@@ -189,42 +175,40 @@ def process_changes_chunk(self, changes_chunk):
         return retry_changes, error_changes
 
 
-def send_to_elasticsearch(index_info, doc_type, doc_id, es_getter, name, data=None,
-                          delete=False, es_merge_update=False):
+def send_to_elasticsearch(index_info=None, doc_type=None, doc_id=None, es_getter=None, name=None, adapter=None,
```

Comment:
Making all the args optional is a bit dangerous since it's not clear what combination of args to provide.

I would suggest creating separate functions with different signatures which call a private function under the hood.

Reply:
https://github.com/dimagi/commcare-hq/pull/32660/commits/028e281a5a0a081e1254ed65b34a622b4de340b1 I did a quick search around in codebase and I realised that its usage has already been refactored. So I removed the params that are no longer needed and marked the essential args non-optional.